application-services/custom/camera-management/appcamera/evam.go

```diff
@@ -0,0 +1,220 @@
+		Name:    pipelineName,
+		Version: pipelineVersion,
+		Profile: profileToken,
+	}
+	var res interface{}
+	baseUrl, err := url.Parse(app.config.AppCustom.EvamBaseUrl)
+	if err != nil {
+		return err
+	}
+	reqPath := path.Join("pipelines", info.Name, info.Version)
+
+	if err = issuePostRequest(context.Background(), &res, baseUrl.String(), reqPath, body); err != nil {
+		return errors.Wrap(err, "POST request to start EVAM pipeline failed")
+	}
+	info.Id = fmt.Sprintf("%v", res)
+
+	if err = app.addPipelineInfo(deviceName, info); err != nil {
+		return err
+	}
+
+	app.lc.Infof("Successfully started EVAM pipeline for the device %s", deviceName)
+	app.lc.Infof("View inference results at 'rtsp://<SYSTEM_IP_ADDRESS>:8554/%s'", deviceName)
+
+	return nil
+}
+
+func (app *CameraManagementApp) stopPipeline(deviceName string) error {
+	var res interface{}
+
+	if info, found := app.getPipelineInfo(deviceName); found {
+		pipelineUrl, err := info.getPipelineUrl(app.config.AppCustom.EvamBaseUrl)
+		if err != nil {
+			return err
+		}
+		if err := issueDeleteRequest(context.Background(), &res, pipelineUrl, info.Id); err != nil {
+			return errors.Wrap(err, "DELETE request to stop EVAM pipeline failed")
+		}
+		app.deletePipelineInfo(deviceName)
+		app.lc.Infof("Successfully stopped EVAM pipeline for the device %s", deviceName)
+	}
+
+	return nil
+}
+
+func (app *CameraManagementApp) createPipelineRequestBody(streamUri string, deviceName string) ([]byte, error) {
+	uri, err := url.Parse(streamUri)
+	if err != nil {
+		return nil, err
+	}
+
+	if creds, err := app.tryGetCredentials(); err != nil {
+		app.lc.Warnf("Error retrieving %s secret from the SecretStore: %s", CameraCredentials, err.Error())
+	} else {
+		uri.User = url.UserPassword(creds.Username, creds.Password)
+	}
+
+	pipelineData := PipelineRequest{
+		Source: Source{
+			URI:  uri.String(),
+			Type: "uri",
+		},
+		Destination: Destination{
+			Metadata: Metadata{
+				Type:  "mqtt",
+				Host:  app.config.AppCustom.MqttAddress,
+				Topic: app.config.AppCustom.MqttTopic,
+			},
+			Frame: Frame{
+				Type: "rtsp",
+				Path: deviceName,
+			},
+		},
+	}
+
+	pipeline, err := json.Marshal(pipelineData)
+	if err != nil {
+		return pipeline, err
+	}
+
+	return pipeline, nil
+}
+
+func (app *CameraManagementApp) getPipelineStatus(deviceName string) (interface{}, error) {
+	if info, found := app.getPipelineInfo(deviceName); found {
+		var res interface{}
+		pipelineUrl, err := info.getPipelineUrl(app.config.AppCustom.EvamBaseUrl)
+		if err != nil {
+			return nil, err
+		}
+		if err := issueGetRequest(context.Background(), &res, pipelineUrl, info.Id+"/status"); err != nil {
+			return nil, errors.Wrap(err, "GET request to query EVAM pipeline failed")
+		}
+		return res, nil
+	}
+
+	return nil, nil
+}
+
+func (app *CameraManagementApp) getAllPipelineStatuses() (map[string]PipelineInfoStatus, error) {
+	res := make(map[string]PipelineInfoStatus)
+
+	for camera, info := range app.pipelinesMap {
```

Comment:
Yes, it appears that would be needed here. However, I'm thinking a blanket RLock around the whole func would be bad since it includes an http request. I will look into locking as part of a pre-fill step and then unlocking before the http calls are made.